packages/typegpu/src/data/compiledIO.ts

```diff
@@ -0,0 +1,149 @@
+import { roundUp } from '../mathUtils';
+import type { Infer } from '../shared/repr';
+import { alignmentOf } from './alignmentOf';
+import { isDisarray, isUnstruct } from './dataTypes';
+import { offsetsForProps } from './offsets';
+import { sizeOf } from './sizeOf';
+import * as wgsl from './wgslTypes';
+
+export let EVAL_ALLOWED_IN_ENV: boolean;
+
+try {
+  new Function('return true');
+  EVAL_ALLOWED_IN_ENV = true;
+} catch {
+  EVAL_ALLOWED_IN_ENV = false;
+}
```

Comment:
To ensure that we're getting the computed value of `EVAL_ALLOWED_IN_ENV` in other files, let's try something like this.

```suggestion
export const EVAL_ALLOWED_IN_ENV: boolean = (() => {
  try {
    new Function('return true');
    return true;
  } catch {
    // Expected error
  }
  return false;
})();
```

Reply:
I can foresee a different behavior in CJS and ESM, which would be hard to catch.